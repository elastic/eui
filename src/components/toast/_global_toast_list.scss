/**
 * 1. Allow list to expand as items are added, but cap it at the screen height.
 * 2. Only show the scroll on hover. Generally, scrolling is bad for toasts.
 * 3. Allow some padding if a scroll shows up.
 */
.euiGlobalToastList {
  @include euiScrollBar;

  display: flex;
  flex-direction: column;
  align-items: stretch;
  position: fixed;
  z-index: $euiZToastList;
  bottom: 0;
  right: 0;
  width: $euiToastWidth + $euiSize + $euiSizeXL; /* 3 */
  padding-right: $euiSize;
  padding-left: $euiSizeXL;
  max-height: 100vh; /* 1 */

  &:hover {
    overflow-y: auto; /* 2 */
  }
}

.euiGlobalToastListItem {
  margin-bottom: $euiSize;
  margin-right: $euiSize;
  animation: $euiAnimSpeedNormal euiShowToast $euiAnimSlightResistance;
  opacity: 1;

  /**
   * 1. justify-content: flex-end interferes with overflowing content, so we'll use this to push
   *    items to the bottom instead.
   */
  &:first-child {
    margin-top: auto; /* 1 */
  }

  &.euiGlobalToastListItem-isDismissed {
    transition: opacity $euiAnimSpeedNormal;
    opacity: 0;
  }
}

@keyframes euiShowToast {
  from {
    transform: translateY($euiSizeL) scale(.9);
    opacity: 0;
  }

  to {
    transform: translateY(0) scale(1);
    opacity: 1;
  }
}

@include euiBreakpoint('xs', 's') {
  /**
   * 1. Mobile we make these 100%. Matching change happens on the item as well.
   */
  .euiGlobalToastList {
    left: 0;
    padding-left: $euiSize;
    width: 100%; /* 1 */
  }
}
