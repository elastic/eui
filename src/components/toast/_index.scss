$euiToastWidth: $euiSize * 20;

@import 'toast';
