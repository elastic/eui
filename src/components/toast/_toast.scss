.euiToast {
  @include euiBottomShadow;

  position: relative;
  padding: $euiSize;
  background-color: $euiColorEmptyShade;
  border: $euiBorderThin;
  width: 100%;

  &:hover .euiToast__closeButton,
  &:focus .euiToast__closeButton, {
    opacity: 1;
  }
}

  /**
   * 1. Fit button to icon.
   */
  .euiToast__closeButton {
    position: absolute;
    top: $euiSize;
    right: $euiSize;
    line-height: 0; /* 1 */
    appearance: none;
    opacity: 0;
    transition: opacity $euiAnimSpeedFast $euiAnimSlightResistance;

    svg {
      fill: tintOrShade($euiTitleColor, 50%, 70%);
    }

    &:hover {
      svg {
        fill: $euiTitleColor;
      }
    }

    &:focus {
      background-color: $euiFocusBackgroundColor;
      opacity: 1;

      svg {
        fill: $euiColorPrimary;
      }
    }
  }

// Modifier naming and colors.
$toastTypes: (
  primary: $euiColorPrimary,
  success: $euiColorSecondary,
  warning: $euiColorWarning,
  danger: $euiColorDanger,
);

// Create button modifiders based upon the map.
@each $name, $color in $toastTypes {
  .euiToast--#{$name} {
    border-top: 2px solid $color;
  }
}

/**
 * 1. Align icon with first line of title text if it wraps.
 * 2. Apply margin to all but last item in the flex.
 * 3. Account for close button.
 */
.euiToastHeader {
  @include euiFontSize;
  padding-right: $euiSizeL; /* 3 */

  display: flex;
  align-items: baseline; /* 1 */

  > * + * {
    margin-left: $euiSizeS; /* 2 */
  }
}

  /**
   * 1. Vertically center icon with first line of title.
   */
  .euiToastHeader__icon {
    flex: 0 0 auto;
    fill: $euiTitleColor;
    transform: translateY(2px); /* 1 */
  }

  .euiToastHeader__title {
    color: $euiTitleColor;
    font-weight: $euiFontWeightLight;
  }

.euiToastHeader--withBody {
  margin-bottom: $euiSizeM;
}
