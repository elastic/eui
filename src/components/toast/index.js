export {
  EuiToast,
} from './toast';

export {
  EuiGlobalToastList,
} from './global_toast_list';

export {
  EuiGlobalToastListItem,
} from './global_toast_list_item';
