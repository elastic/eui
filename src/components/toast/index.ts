/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

export { EuiToast } from './toast';

export {
  EuiGlobalToastList,
  EuiGlobalToastListProps,
  Toast as EuiGlobalToastListToast,
} from './global_toast_list';

export {
  EuiGlobalToastListItem,
  EuiGlobalToastListItemProps,
} from './global_toast_list_item';
