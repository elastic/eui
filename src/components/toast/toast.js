import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import {
  ICON_TYPES,
  EuiIcon,
} from '../icon';

import {
  EuiText,
} from '../text';

const colorToClassNameMap = {
  primary: 'euiToast--primary',
  success: 'euiToast--success',
  warning: 'euiToast--warning',
  danger: 'euiToast--danger',
};

export const COLORS = Object.keys(colorToClassNameMap);

export const EuiToast = ({ title, color, iconType, onClose, children, className, ...rest }) => {
  const classes = classNames('euiToast', colorToClassNameMap[color], className);
  const headerClasses = classNames('euiToastHeader', {
    'euiToastHeader--withBody': children,
  });

  let headerIcon;

  if (iconType) {
    headerIcon = (
      <EuiIcon
        className="euiToastHeader__icon"
        type={iconType}
        size="m"
        aria-hidden="true"
      />
    );
  }

  let closeButton;

  if (onClose) {
    closeButton = (
      <button
        type="button"
        className="euiToast__closeButton"
        aria-label="Dismiss toast"
        onClick={onClose}
        data-test-subj="toastCloseButton"
      >
        <EuiIcon
          type="cross"
          size="m"
          aria-hidden="true"
        />
      </button>
    );
  }

  let optionalBody;

  if (children) {
    optionalBody = (
      <EuiText size="s" className="euiToastBody">
        {children}
      </EuiText>
    );
  }

  return (
    <div
      className={classes}
      {...rest}
    >
      <div className={headerClasses}>
        {headerIcon}

        <span className="euiToastHeader__title">
          {title}
        </span>
      </div>

      {closeButton}
      {optionalBody}
    </div>
  );
};

EuiToast.propTypes = {
  title: PropTypes.node,
  iconType: PropTypes.oneOf(ICON_TYPES),
  color: PropTypes.oneOf(COLORS),
  onClose: PropTypes.func,
  children: PropTypes.node,
};
