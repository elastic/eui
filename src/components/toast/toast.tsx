/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React, {
  FunctionComponent,
  HTMLAttributes,
  ReactElement,
  ReactNode,
} from 'react';
import classNames from 'classnames';

import { CommonProps, keysOf } from '../common';
import { EuiScreenReaderOnly } from '../accessibility';
import { EuiI18n } from '../i18n';

import { IconType, EuiIcon } from '../icon';

import { EuiText } from '../text';

type ToastColor = 'primary' | 'success' | 'warning' | 'danger';

const colorToClassNameMap: { [color in ToastColor]: string } = {
  primary: 'euiToast--primary',
  success: 'euiToast--success',
  warning: 'euiToast--warning',
  danger: 'euiToast--danger',
};

export const COLORS = keysOf(colorToClassNameMap);

export interface EuiToastProps
  extends CommonProps,
    Omit<HTMLAttributes<HTMLDivElement>, 'title'> {
  title?: ReactNode;
  color?: ToastColor;
  iconType?: IconType;
  onClose?: () => void;
}

export const EuiToast: FunctionComponent<EuiToastProps> = ({
  title,
  color,
  iconType,
  onClose,
  children,
  className,
  ...rest
}) => {
  const classes = classNames(
    'euiToast',
    color ? colorToClassNameMap[color] : null,
    className
  );
  const headerClasses = classNames('euiToastHeader', {
    'euiToastHeader--withBody': children,
  });

  let headerIcon: ReactElement;

  if (iconType) {
    headerIcon = (
      <EuiIcon
        className="euiToastHeader__icon"
        type={iconType}
        size="m"
        aria-hidden="true"
      />
    );
  }

  let closeButton;

  if (onClose) {
    closeButton = (
      <EuiI18n token="euiToast.dismissToast" default="Dismiss toast">
        {(dismissToast: string) => (
          <button
            type="button"
            className="euiToast__closeButton"
            aria-label={dismissToast}
            onClick={onClose}
            data-test-subj="toastCloseButton">
            <EuiIcon type="cross" size="m" aria-hidden="true" />
          </button>
        )}
      </EuiI18n>
    );
  }

  let optionalBody;

  if (children) {
    optionalBody = (
      <EuiText size="s" className="euiToastBody">
        {children}
      </EuiText>
    );
  }

  return (
    <div className={classes} {...rest}>
      <EuiScreenReaderOnly>
        <p>
          <EuiI18n
            token="euiToast.newNotification"
            default="A new notification appears"
          />
        </p>
      </EuiScreenReaderOnly>

      <EuiI18n token="euiToast.notification" default="Notification">
        {(notification: string) => (
          <div
            className={headerClasses}
            aria-label={notification}
            data-test-subj="euiToastHeader">
            {headerIcon}

            <span className="euiToastHeader__title">{title}</span>
          </div>
        )}
      </EuiI18n>

      {closeButton}
      {optionalBody}
    </div>
  );
};
