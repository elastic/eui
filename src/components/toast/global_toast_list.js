import React, {
  Component,
} from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { Timer } from '../../services/time';
import { EuiGlobalToastListItem } from './global_toast_list_item';
import { EuiToast } from './toast';

export const TOAST_FADE_OUT_MS = 250;

export class EuiGlobalToastList extends Component {
  constructor(props) {
    super(props);

    this.state = {
      toastIdToDismissedMap: {},
    };

    this.dismissTimeoutIds = [];
    this.toastIdToTimerMap = {};

    this.isScrollingToBottom = false;
    this.isScrolledToBottom = true;
  }

  static propTypes = {
    className: PropTypes.string,
    toasts: PropTypes.array,
    dismissToast: PropTypes.func.isRequired,
    toastLifeTimeMs: PropTypes.number.isRequired,
  };

  static defaultProps = {
    toasts: [],
  };

  startScrollingToBottom() {
    this.isScrollingToBottom = true;

    const scrollToBottom = () => {
      const position = this.listElement.scrollTop;
      const destination = this.listElement.scrollHeight - this.listElement.clientHeight;
      const distanceToDestination = destination - position;

      if (distanceToDestination < 5) {
        this.listElement.scrollTop = destination;
        this.isScrollingToBottom = false;
        this.isScrolledToBottom = true;
        return;
      }

      this.listElement.scrollTop = position + distanceToDestination * 0.25;

      if (this.isScrollingToBottom) {
        window.requestAnimationFrame(scrollToBottom);
      }
    };

    window.requestAnimationFrame(scrollToBottom);
  }

  onMouseEnter = () => {
    // Stop scrolling to bottom if we're in mid-scroll, because the user wants to interact with
    // the list.
    this.isScrollingToBottom = false;
    this.isUserInteracting = true;

    // Don't let toasts dismiss themselves while the user is interacting with them.
    for (const toastId in this.toastIdToTimerMap) {
      if (this.toastIdToTimerMap.hasOwnProperty(toastId)) {
        const timer = this.toastIdToTimerMap[toastId];
        timer.pause();
      }
    }
  };

  onMouseLeave = () => {
    this.isUserInteracting = false;
    for (const toastId in this.toastIdToTimerMap) {
      if (this.toastIdToTimerMap.hasOwnProperty(toastId)) {
        const timer = this.toastIdToTimerMap[toastId];
        timer.resume();
      }
    }
  };

  onScroll = () => {
    this.isScrolledToBottom =
      this.listElement.scrollHeight - this.listElement.scrollTop === this.listElement.clientHeight;
  };

  scheduleAllToastsForDismissal = () => {
    this.props.toasts.forEach(toast => {
      if (!this.toastIdToTimerMap[toast.id]) {
        this.scheduleToastForDismissal(toast);
      }
    });
  };

  scheduleToastForDismissal = (toast) => {
    // Start fading the toast out once its lifetime elapses.
    this.toastIdToTimerMap[toast.id] =
      new Timer(this.dismissToast.bind(this, toast), this.props.toastLifeTimeMs);
  };

  dismissToast = (toast) => {
    // Remove the toast after it's done fading out.
    this.dismissTimeoutIds.push(setTimeout(() => {
      this.props.dismissToast(toast);
      this.toastIdToTimerMap[toast.id].clear();
      delete this.toastIdToTimerMap[toast.id];

      this.setState(prevState => {
        const toastIdToDismissedMap = { ...prevState.toastIdToDismissedMap };
        delete toastIdToDismissedMap[toast.id];

        return {
          toastIdToDismissedMap,
        };
      });
    }, TOAST_FADE_OUT_MS));

    this.setState(prevState => {
      const toastIdToDismissedMap = {
        ...prevState.toastIdToDismissedMap,
        [toast.id]: true,
      };

      return {
        toastIdToDismissedMap,
      };
    });
  };

  componentDidMount() {
    this.listElement.addEventListener('scroll', this.onScroll);
    this.listElement.addEventListener('mouseenter', this.onMouseEnter);
    this.listElement.addEventListener('mouseleave', this.onMouseLeave);
    this.scheduleAllToastsForDismissal();
  }

  componentDidUpdate(prevProps) {
    this.scheduleAllToastsForDismissal();

    if (!this.isUserInteracting) {
      // If the user has scrolled up the toast list then we don't want to annoy them by scrolling
      // all the way back to the bottom.
      if (this.isScrolledToBottom) {
        if (prevProps.toasts.length < this.props.toasts.length) {
          this.startScrollingToBottom();
        }
      }
    }
  }

  componentWillUnmount() {
    this.listElement.removeEventListener('scroll', this.onScroll);
    this.listElement.removeEventListener('mouseenter', this.onMouseEnter);
    this.listElement.removeEventListener('mouseleave', this.onMouseLeave);
    this.dismissTimeoutIds.forEach(clearTimeout);
    for (const toastId in this.toastIdToTimerMap) {
      if (this.toastIdToTimerMap.hasOwnProperty(toastId)) {
        const timer = this.toastIdToTimerMap[toastId];
        timer.clear();
      }
    }
  }

  render() {
    const {
      className,
      toasts,
      dismissToast, // eslint-disable-line no-unused-vars
      toastLifeTimeMs, // eslint-disable-line no-unused-vars
      ...rest
    } = this.props;

    const renderedToasts = toasts.map(toast => {
      const {
        text,
        ...rest
      } = toast;

      return (
        <EuiGlobalToastListItem
          key={toast.id}
          isDismissed={this.state.toastIdToDismissedMap[toast.id]}
        >
          <EuiToast
            onClose={this.dismissToast.bind(this, toast)}
            {...rest}
          >
            {text}
          </EuiToast>
        </EuiGlobalToastListItem>
      );
    });

    const classes = classNames('euiGlobalToastList', className);

    return (
      <div
        ref={element => { this.listElement = element; }}
        className={classes}
        {...rest}
      >
        {renderedToasts}
      </div>
    );
  }
}
