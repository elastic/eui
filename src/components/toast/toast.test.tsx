/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React from 'react';
import { render, mount } from 'enzyme';
import { findTestSubject, requiredProps } from '../../test';

import { COLORS, EuiToast } from './toast';

describe('EuiToast', () => {
  test('is rendered', () => {
    const component = render(
      <EuiToast {...requiredProps} title="test title">
        <p>Hi</p>
      </EuiToast>
    );

    expect(component).toMatchSnapshot();
  });

  describe('Props', () => {
    describe('title', () => {
      test('is rendered', () => {
        const component = <EuiToast title="toast title" />;
        expect(mount(component)).toMatchSnapshot();
      });
    });

    describe('color', () => {
      COLORS.forEach((color) => {
        test(`${color} is rendered`, () => {
          const component = <EuiToast color={color} title="test title" />;
          expect(mount(component)).toMatchSnapshot();
        });
      });
    });

    describe('iconType', () => {
      test('is rendered', () => {
        const component = <EuiToast iconType="user" title="test title" />;
        expect(mount(component)).toMatchSnapshot();
      });
    });

    describe('onClose', () => {
      test('is called when the close button is clicked', () => {
        const onCloseHandler = jest.fn();

        const component = mount(
          <EuiToast onClose={onCloseHandler} title="test title" />
        );
        const closeButton = findTestSubject(component, 'toastCloseButton');
        closeButton.simulate('click');

        expect(onCloseHandler).toBeCalledTimes(1);
      });
    });
  });
});
