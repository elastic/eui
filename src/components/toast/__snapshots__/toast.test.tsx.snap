// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EuiToast Props color danger is rendered 1`] = `
<EuiToast
  color="danger"
  title="test title"
>
  <div
    className="euiToast euiToast--danger"
  >
    <EuiScreenReaderOnly>
      <p
        className="euiScreenReaderOnly"
      >
        <EuiI18n
          default="A new notification appears"
          token="euiToast.newNotification"
        >
          A new notification appears
        </EuiI18n>
      </p>
    </EuiScreenReaderOnly>
    <EuiI18n
      default="Notification"
      token="euiToast.notification"
    >
      <div
        aria-label="Notification"
        className="euiToastHeader"
        data-test-subj="euiToastHeader"
      >
        <span
          className="euiToastHeader__title"
        >
          test title
        </span>
      </div>
    </EuiI18n>
  </div>
</EuiToast>
`;

exports[`EuiToast Props color primary is rendered 1`] = `
<EuiToast
  color="primary"
  title="test title"
>
  <div
    className="euiToast euiToast--primary"
  >
    <EuiScreenReaderOnly>
      <p
        className="euiScreenReaderOnly"
      >
        <EuiI18n
          default="A new notification appears"
          token="euiToast.newNotification"
        >
          A new notification appears
        </EuiI18n>
      </p>
    </EuiScreenReaderOnly>
    <EuiI18n
      default="Notification"
      token="euiToast.notification"
    >
      <div
        aria-label="Notification"
        className="euiToastHeader"
        data-test-subj="euiToastHeader"
      >
        <span
          className="euiToastHeader__title"
        >
          test title
        </span>
      </div>
    </EuiI18n>
  </div>
</EuiToast>
`;

exports[`EuiToast Props color success is rendered 1`] = `
<EuiToast
  color="success"
  title="test title"
>
  <div
    className="euiToast euiToast--success"
  >
    <EuiScreenReaderOnly>
      <p
        className="euiScreenReaderOnly"
      >
        <EuiI18n
          default="A new notification appears"
          token="euiToast.newNotification"
        >
          A new notification appears
        </EuiI18n>
      </p>
    </EuiScreenReaderOnly>
    <EuiI18n
      default="Notification"
      token="euiToast.notification"
    >
      <div
        aria-label="Notification"
        className="euiToastHeader"
        data-test-subj="euiToastHeader"
      >
        <span
          className="euiToastHeader__title"
        >
          test title
        </span>
      </div>
    </EuiI18n>
  </div>
</EuiToast>
`;

exports[`EuiToast Props color warning is rendered 1`] = `
<EuiToast
  color="warning"
  title="test title"
>
  <div
    className="euiToast euiToast--warning"
  >
    <EuiScreenReaderOnly>
      <p
        className="euiScreenReaderOnly"
      >
        <EuiI18n
          default="A new notification appears"
          token="euiToast.newNotification"
        >
          A new notification appears
        </EuiI18n>
      </p>
    </EuiScreenReaderOnly>
    <EuiI18n
      default="Notification"
      token="euiToast.notification"
    >
      <div
        aria-label="Notification"
        className="euiToastHeader"
        data-test-subj="euiToastHeader"
      >
        <span
          className="euiToastHeader__title"
        >
          test title
        </span>
      </div>
    </EuiI18n>
  </div>
</EuiToast>
`;

exports[`EuiToast Props iconType is rendered 1`] = `
<EuiToast
  iconType="user"
  title="test title"
>
  <div
    className="euiToast"
  >
    <EuiScreenReaderOnly>
      <p
        className="euiScreenReaderOnly"
      >
        <EuiI18n
          default="A new notification appears"
          token="euiToast.newNotification"
        >
          A new notification appears
        </EuiI18n>
      </p>
    </EuiScreenReaderOnly>
    <EuiI18n
      default="Notification"
      token="euiToast.notification"
    >
      <div
        aria-label="Notification"
        className="euiToastHeader"
        data-test-subj="euiToastHeader"
      >
        <EuiIcon
          aria-hidden="true"
          className="euiToastHeader__icon"
          size="m"
          type="user"
        >
          <div
            aria-hidden="true"
            className="euiToastHeader__icon"
            data-euiicon-type="user"
            size="m"
          />
        </EuiIcon>
        <span
          className="euiToastHeader__title"
        >
          test title
        </span>
      </div>
    </EuiI18n>
  </div>
</EuiToast>
`;

exports[`EuiToast Props title is rendered 1`] = `
<EuiToast
  title="toast title"
>
  <div
    className="euiToast"
  >
    <EuiScreenReaderOnly>
      <p
        className="euiScreenReaderOnly"
      >
        <EuiI18n
          default="A new notification appears"
          token="euiToast.newNotification"
        >
          A new notification appears
        </EuiI18n>
      </p>
    </EuiScreenReaderOnly>
    <EuiI18n
      default="Notification"
      token="euiToast.notification"
    >
      <div
        aria-label="Notification"
        className="euiToastHeader"
        data-test-subj="euiToastHeader"
      >
        <span
          className="euiToastHeader__title"
        >
          toast title
        </span>
      </div>
    </EuiI18n>
  </div>
</EuiToast>
`;

exports[`EuiToast is rendered 1`] = `
<div
  aria-label="aria-label"
  class="euiToast testClass1 testClass2"
  data-test-subj="test subject string"
>
  <p
    class="euiScreenReaderOnly"
  >
    A new notification appears
  </p>
  <div
    aria-label="Notification"
    class="euiToastHeader euiToastHeader--withBody"
    data-test-subj="euiToastHeader"
  >
    <span
      class="euiToastHeader__title"
    >
      test title
    </span>
  </div>
  <div
    class="euiText euiText--small euiToastBody"
  >
    <p>
      Hi
    </p>
  </div>
</div>
`;
